### DISCUSSION TITLE
dosemu2-2.0pre9 revision 7076: Assertion `syncpipe[0] < MAX_FD' failed.

### DISCUSSION THREAD

**Discussion by ghost** [dosemu2/dosemu2#1996] [General]
I've actually been seeing this issue for a little while.  Currently, after my system has been running for a while, I start seeing this error when dosemu2 runs:

`dosemu.bin: ./build/../src/base/misc/ioctl.c:305: void ioselect_init(void): Assertion `syncpipe[0] < MAX_FD' failed.ERROR: cpu exception in dosemu code outside of VM86()!Aborted (Signal sent by tkill() 376717 1000)ERROR: Please install gdb, update dosemu from git, compile it with debuginfo and make a bug report with the content of ~/.dosemu/boot.log athttps://github.com/dosemu2/dosemu2/issuesPlease provide any additional info you can, like the test-cases,URLs and all the rest that fits.`

Rebooting my system seems to fix it, until this happens again.

**Comment by stsp** [comment]
Can you be more specific?
What you do to get this error?
Does it happen on dosemu startup?
How exactly are you starting it?
Have you checked /proc/self/fd before
starting dosemu, maybe there are open
files leaked?
